### PR TITLE
feat(store): add mongodb store backend to support meta service

### PR DESCRIPTION
**Pull request by ryantd**

### CODE REVIEW

**Comment by ryantd** [comment]
@MarvinMiao it's ready to have a review, and the current CR includes MongoDB store logic code and unit test

File: flink-ai-flow/ai_flow/metadata_store/service/service.py [hunk 1]
```diff
@@ -33,12 +33,18 @@
...
+from ai_flow.store.mongo_store import MongoStore
+from ai_flow.store.db.db_util import extract_db_engine_from_uri
+from ai_flow.application_master.master_config import DBType
 
 
 class MetadataService(metadata_service_pb2_grpc.MetadataServiceServicer):
     def __init__(self, db_uri, server_uri):
-        db_uri = db_uri
-        self.store = SqlAlchemyStore(db_uri)
+        db_engine = extract_db_engine_from_uri(db_uri)
```
Comment (ryantd) [inline flink-ai-flow/ai_flow/metadata_store/service/service.py:43]: 是否需要更优雅的方式？

File: flink-ai-flow/ai_flow/metadata_store/utils/MetaToTable.py [hunk 2]
```diff
@@ -41,15 +45,19 @@ def example_meta_to_table(name, support_type, data_type, data_format, descriptio
...
+        return _class(name=name, support_type=support_type, data_type=data_type, format=data_format,
+                      description=description, batch_uri=batch_uri, stream_uri=stream_uri, create_time=create_time,
+                      update_time=update_time, properties=properties, name_list=name_list, type_list=data_type_list,
+                      catalog_name=catalog_name, catalog_type=catalog_type, catalog_database=catalog_database,
+                      catalog_connection_uri=catalog_connection_uri, catalog_version=catalog_version,
+                      catalog_table=catalog_table)
 
     @staticmethod
-    def example_meta_list_to_table(example_meta_list: List[ExampleMeta]):
+    def example_meta_list_to_table(example_meta_list: List[ExampleMeta], store_type='SqlAlchemyStore'):
```
Comment (ryantd) [inline flink-ai-flow/ai_flow/metadata_store/utils/MetaToTable.py:60]: 有部分表的新建逻辑不在store内，这里需要传入一个标识，以便区分sql还是mongo

File: flink-ai-flow/ai_flow/store/db/db_model.py [hunk 3]
```diff
@@ -375,3 +378,429 @@ class SqlMember(base):
     def __repr__(self):
         return '<SqlMember ({}, {}, {}, {}, {})>'.format(
             self.id, self.version, self.server_uri, self.update_time, self.uuid)
+
+
+class DocumentExample(Document):
```
Comment (ryantd) [inline flink-ai-flow/ai_flow/store/db/db_model.py]: 按如此取名，自动新建的collection会带有"document"前缀，是否需要把sql和mongo放到两个文件下；然后此处改为Example(Document)

File: flink-ai-flow/ai_flow/store/db/db_model.py [hunk 4]
```diff
@@ -375,3 +378,429 @@ class SqlMember(base):
...
+    model_name = StringField(max_length=255, required=True)
+    model_version = StringField(max_length=10, required=True)
+    model_path = StringField(max_length=500, default=None)
+    model_metric = StringField(max_length=500, default=None)
+    model_flavor = StringField(max_length=500, default=None)
+    version_desc = StringField(max_length=1000)
+    version_status = StringField(max_length=20,
+                                 default=ModelVersionStatus.to_string(ModelVersionStatus.READY))
+    current_stage = StringField(max_length=20, default=STAGE_GENERATED)
+    name_version_current_stage_unique = StringField(max_length=1000, required=True, unique=True)
```
Comment (ryantd) [inline flink-ai-flow/ai_flow/store/db/db_model.py:608]: 通过新建一个字段来适配sql的UNIQUE(name, version, current_stage)

File: flink-ai-flow/ai_flow/metadata_store/service/service.py (hunk 1 above)
Comment (Gforky) [reply flink-ai-flow/ai_flow/metadata_store/service/service.py:43]: mongodb的连接方式可以考虑使用connect函数配合keyword参数的方式，直接使用db_uri可能解释性不强

File: flink-ai-flow/ai_flow/metadata_store/utils/MetaToTable.py (hunk 2 above)
Comment (Gforky) [reply flink-ai-flow/ai_flow/metadata_store/utils/MetaToTable.py:60]: agree

File: flink-ai-flow/ai_flow/store/db/db_model.py (hunk 3 above)
Comment (Gforky) [reply flink-ai-flow/ai_flow/store/db/db_model.py]: maybe change to MongoExample?

**Review by Gforky — COMMENTED** [review]
1. no need to implement `DocumentEvent`, already been implemented in mongo_notification.py as `MongoEvent`.
2. why delete test_sqlalchemy_store.py?

File: flink-ai-flow/ai_flow/store/db/db_model.py [hunk 5]
```diff
@@ -375,3 +378,429 @@ class SqlMember(base):
...
+    def to_detail_entity(self):
+        latest_version = None
+        for model_version in reversed(self.model_version):
+            if model_version.current_stage != STAGE_DELETED:
+                latest_version = model_version.to_meta_entity()
+                break
+        return RegisteredModelDetail(self.model_name, self.model_type, self.model_desc, latest_version)
+
+
+class DocumentEvent(Document):
```
Comment (ryantd) [inline flink-ai-flow/ai_flow/store/db/db_model.py]: 删除

File: flink-ai-flow/ai_flow/metadata_store/service/service.py (hunk 1 above)
Comment (ryantd) [reply flink-ai-flow/ai_flow/metadata_store/service/service.py:43]: mongo均改为key value形式

**Review by Gforky — APPROVED** [review]
agree, ready to merge

**Pull request merged**